### PR TITLE
feat(node): added rootDir flag when generating node/lib

### PR DESCRIPTION
**Pull request by mandarini**
## Current Behavior
`rootDir` is ignored when added in `tsconfig.lib.json` in node generated libraries. Only `srcRootForCompilationRoot` in `workspace.json` is taken into account.

## Expected Behavior
`rootDir` is taken into account when added in `tsconfig.lib.json` in node generated libraries. `srcRootForCompilationRoot` in `workspace.json` takes precedence.

## Related Issue(s)
#3625

Fixes #
#3625

### CODE REVIEW

**Comment by mandarini** [comment]
This still has an issue that needs to be fixed, in the build folder structure.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@nrwl/nx-source",
-  "version": "10.1.0",
+  "version": "999.5.9",
```
Comment (Cammisuli) [inline package.json:3]: @mandarini you should revert this change 😉

**Review by Cammisuli — APPROVED** [review]
